### PR TITLE
IDEMPIERE-5331 Create Production from Order (all Lines)

### PR DESCRIPTION
**Pull request by CarlosRuiz-globalqss**
https://idempiere.atlassian.net/browse/IDEMPIERE-5331

### CODE REVIEW

File: org.adempiere.base.process/src/org/compiere/process/OrderCreateProduction.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+			production.setProductionQty(line.getQtyOrdered().subtract(line.getQtyDelivered()));
+			production.setDatePromised(line.getDatePromised());
+			if (product.getM_Locator_ID() > 0)
+				production.setM_Locator_ID(product.getM_Locator_ID());
+			production.setC_OrderLine_ID(line.getC_OrderLine_ID());
+
+			int locator = product.getM_Locator_ID();
+			if (locator == 0)
+				locator = MWarehouse.get(getCtx(), line.getM_Warehouse_ID()).getDefaultLocator().get_ID();
+			production.setM_Locator_ID(locator);
```
Comment (hengsin) [inline org.adempiere.base.process/src/org/compiere/process/OrderCreateProduction.java:108]: this make line 101 & 102 above redundant

Comment (CarlosRuiz-globalqss) [reply org.adempiere.base.process/src/org/compiere/process/OrderCreateProduction.java:106]: Thanks @hengsin - I copied the code from OrderLineCreateProduction and didn't notice the redundant assignment there.

So, with the last commit I refactored also that class and additionally removed the unused parameters

**Review by hengsin — CHANGES_REQUESTED** [review]
1. Tested order with PChair. Production document created with 9 production line which is not correct.
2. Failed the following scenario
- Create and Complete Sales Order with PChair
- Run "Create Production from Order"
- Void the production document created.
- Goes back order, run "Create Production from Order Line". The process error out with "Production has already been created: xxx"

**Comment by CarlosRuiz-globalqss** [comment]
Thanks @hengsin 

> Tested order with PChair. Production document created with 9 production line which is not correct.

This is not changed in this ticket, it's the actual behavior of create production, if you create a production with PChair manually, it's created with 9 lines, I see it's doing BOM explosion for the legs.
I think to discuss if that's correct or not - or if it must be made configurable, requires a different ticket.

>   Failed the following scenario
>    Create and Complete Sales Order with PChair
>    Run "Create Production from Order"
>    Void the production document created.
>    Goes back order, run "Create Production from Order Line". The process error out with "Production has already been created: xxx"

Also not introduced in this ticket, but is a good opportunity to solve it too.  Committed the fix for void and reverse.

**Comment by hengsin** [comment]
Our treatment of phantom bom item is wrong, see https://blogs.sap.com/2008/04/24/what-is-a-phantom-item-how-is-it-used/

Anyway, that can be addressed in a separate ticket.

**Comment by hengsin** [comment]
My mistake, that BOM explosion is actually correct.

**Review by hengsin — APPROVED** [review]
tested

**Pull request merged**